build_rules.gradle

```diff
@@ -0,0 +1,275 @@
+ * License); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+println "Applying build_rules.gradle to $project.name"
+
+/*************************************************************************************************/
+// Apply common properties/repositories and tasks to all builds.
+
+group = "org.apache.beam"
+version = "2.3.0-SNAPSHOT"
+
+// Define the default set of properties
+repositories {
+  mavenLocal()
+  mavenCentral()
+  jcenter()
+
+  // Release staging repository
+  maven { url "https://oss.sonatype.org/content/repositories/staging/" }
+
+  // Apache nightly snapshots
+  maven { url "https://repository.apache.org/snapshots" }
+
+  // Apache release snapshots
+  maven { url "https://repository.apache.org/content/repositories/releases" }
+}
+
+// Provide code coverage
+apply plugin: "jacoco"
+
+// Provies "htmlDependencyReport" task which spits out a nice page per subproject
+// showing all the dependencies per Gradle configuration.
+apply plugin: "project-report"
+
+// Apply a task dependency visualization plugin which creates a ".dot" file in the build directory
+// giving the task dependencies for that project
+apply plugin: "cz.malohlava.visteg"
+
+/*************************************************************************************************/
+
+/** Used as configuration within the applyJavaNature closure. */
+class JavaNatureConfiguration {
+  double javaVersion = 1.7
+  boolean enableFindbugs = true
+  boolean enableShadow = true
+}
+
+/** Takes a map of parameters. See JavaNatureConfiguration for the set of parameters allowed. */
+ext.applyJavaNature = {
+  println "applyJavaNature with " + (it ? "$it" : "default configuration") + " for project $project.name"
+  // Use the implicit it parameter of the closure to handle zero argument or one argument map calls.
+  JavaNatureConfiguration configuration = it ? it as JavaNatureConfiguration : new JavaNatureConfiguration()
+  apply plugin: "maven"
+  apply plugin: "java"
+
+  sourceCompatibility = configuration.javaVersion
+  targetCompatibility = configuration.javaVersion
+  tasks.withType(JavaCompile) {
+    options.encoding = "UTF-8"
+  }
+
+  test {
+    // Configure the gradle include defaults to be equivalent to the maven-surefire-plugin
+    // http://maven.apache.org/components/surefire/maven-surefire-plugin/test-mojo.html
+    include "**/Test*.class"
+    include "**/*Test.class"
+    include "**/*Tests.class"
+    include "**/*TestCase.class"
+    useJUnit { }
+  }
+
+  apply plugin: "net.ltgt.apt"
```

Comment:
<!--new_thread; commit:8ca3fdca276a74973cd428d7321b995f7b7ebccf; resolved:0-->
These two lines are close to each other - are they linked by groovy syntax or is it just coincidence?